### PR TITLE
feat(theme-default): optimizing css variable acquisition

### PR DESCRIPTION
**Pull request by zhaopan-pan**
optimize the acquisition method of css variables and encapsulate related methods

### CODE REVIEW

File: ecosystem/theme-default/src/client/utils/updateDeviceStatus.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const DeviceTypeMap = {
+  [DeviceType.MOBILE]: Number(cssVars.mobile?.replace('px', '')),
+}
+
+/**
+ * add listener to detect screen though device type
+ */
+export const updateDeviceStatus = (
+  deviceType: DeviceType,
+  callback: (w: number) => void
```
Comment (Mister-Hope) [inline ecosystem/theme-default/src/client/utils/updateDeviceStatus.ts:17]:
```suggestion
  callback: (width: number) => void
```

**Review by Mister-Hope — APPROVED** [review]

**Comment by zhaopan-pan** [comment]
The above checks error seems to be caused by the css-loader of webpack not opening cssMoudle🤔

File: ecosystem/theme-default/src/client/shim.d.ts [hunk 2]
```diff
@@ -3,3 +3,11 @@ declare module '*.vue' {
   const comp: ComponentOptions
   export default comp
 }
+
+declare module '*.module.scss?module' {
+  type Variables = {
+    [className: string]: string
+  }
+  const cssVar: Variables
+  export default cssVar
```
Comment (meteorlxy) [inline ecosystem/theme-default/src/client/shim.d.ts:12]:
```suggestion
  const cssVars: Record<string, string>
  export default cssVars
```

File: ecosystem/theme-default/src/client/utils/updateDeviceStatus.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { onMounted } from 'vue'
+import cssVars from '../styles/_variables.module.scss?module'
+
+export enum DeviceType {
+  MOBILE = 'mobile',
+}
+
+const DeviceTypeMap = {
+  [DeviceType.MOBILE]: Number(cssVars.mobile?.replace('px', '')),
```
Comment (meteorlxy) [inline ecosystem/theme-default/src/client/utils/updateDeviceStatus.ts:9]: `Number.parseInt()` could be better

File: ecosystem/theme-default/src/client/utils/updateDeviceStatus.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * add listener to detect screen though device type
+ */
+export const updateDeviceStatus = (
+  deviceType: DeviceType,
+  callback: (width: number) => void
+): void => {
+  const width = DeviceTypeMap[deviceType]
+  if (!width) return
```
Comment (meteorlxy) [inline ecosystem/theme-default/src/client/utils/updateDeviceStatus.ts:20]:
```suggestion
  if (!Number.isInteger(width)) {
      if (__DEV__) throw new Error('xxxx')
      return
  }
```

**Review by meteorlxy — COMMENTED** [review]
Nice implementation! Suggested some minor changes

File: ecosystem/theme-default/src/client/shim.d.ts (hunk 2 above)
Comment (zhaopan-pan) [reply ecosystem/theme-default/src/client/shim.d.ts]: Here is an easier way, I optimized it

File: ecosystem/theme-default/src/client/utils/updateDeviceStatus.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+const DeviceTypeMap = {
+  [DeviceType.MOBILE]: Number.parseInt(cssVars.mobile?.replace('px', ''), 10),
+}
+
+/**
+ * add listener to detect screen though device type
+ */
+export const updateDeviceStatus = (
```
Comment (meteorlxy) [inline ecosystem/theme-default/src/client/utils/updateDeviceStatus.ts:15]: This method is more like a composable than a util. It's required to be used in component setup. So better to move it under composables dir and rename it with `use` prefix

File: ecosystem/theme-default/src/client/components/Navbar.vue [hunk 6]
```diff
@@ -24,27 +25,20 @@ const linksWrapperStyle = computed(() => {
...
+      navbar.value!.offsetWidth -
+      navbarHorizontalPadding -
+      (navbarBrand.value?.offsetWidth || 0)
   }
-  handleLinksWrapWidth()
-  window.addEventListener('resize', handleLinksWrapWidth, false)
-  window.addEventListener('orientationchange', handleLinksWrapWidth, false)
-})
+}
+updateDeviceStatus(DeviceType.MOBILE, handleLinksWrapWidth)
```
Comment (meteorlxy) [inline ecosystem/theme-default/src/client/components/Navbar.vue:41]: Maybe simplify to `updateDeviceStatus(DeviceType.MOBILE, (...) => { ... })`

Seems that we don't need to declare the callback function separately

**Comment by meteorlxy** [comment]
@zhaopan-pan added 2 more comments

File: ecosystem/theme-default/src/client/utils/updateDeviceStatus.ts (hunk 5 above)
Comment (zhaopan-pan) [reply ecosystem/theme-default/src/client/utils/updateDeviceStatus.ts]: I added it under composites

**Comment by zhaopan-pan** [comment]
> @zhaopan-pan added 2 more comments

i has optimized it

**Review by meteorlxy — APPROVED** [review]

**Comment by meteorlxy** [comment]
Thanks for your contributions! ❤️

**Pull request merged**